src/include/execution/sql/value.h

```diff
@@ -237,38 +179,54 @@ struct StringVal : public Val {
   std::string_view StringView() const {
     TERRIER_ASSERT(!is_null_,
                    "You should be doing a NULL check before attempting to generate a std::string_view of a StringVal.");
-    return std::string_view(Content(), len_);
+    return std::string_view(GetContent(), GetLength());
   }
 
   /**
-   * Helper method to turn a StringVal into a VarlenEntry.
-   * @param str input to be turned into a VarlenEntry
-   * @param own whether the varlen entry to own the string
-   * @return VarlenEntry representing StringVal
+   * @return Threshold for inlining.
+   */
+  static uint32_t InlineThreshold() { return storage::VarlenEntry::InlineThreshold(); }
+
+  /**
+   * Helper method to create a VarlenEntry from a StringVal.
+   * @param str The input to be turned into a VarlenEntry.
+   * @param own Whether the VarlenEntry should own the resulting string.
+   * @return A VarlenEntry with the contents of the StringVal.
    */
   static storage::VarlenEntry CreateVarlen(const StringVal &str, bool own) {
     if (str.is_null_) {
       return terrier::storage::VarlenEntry::CreateInline(static_cast<const terrier::byte *>(nullptr), 0);
     }
-    if (str.len_ > storage::VarlenEntry::InlineThreshold()) {
+    if (str.GetLength() > storage::VarlenEntry::InlineThreshold()) {
       if (own) {
-        byte *contents = common::AllocationUtil::AllocateAligned(str.len_);
-        std::memcpy(contents, str.Content(), str.len_);
-        return terrier::storage::VarlenEntry::Create(contents, str.len_, true);
+        byte *contents = common::AllocationUtil::AllocateAligned(str.GetLength());
```

Comment:
Maybe add a TODO that we may want to be smarter about this allocation in the future. See `value_util.cpp:13` as an example.